libnavigation-core/src/test/java/com/mapbox/navigation/core/lifecycle/CarAppLifecycleOwnerTest.kt

```diff
@@ -38,12 +38,56 @@ class CarAppLifecycleOwnerTest {
             val activity: Activity = mockActivity()
             onActivityCreated(activity, mockk())
             onActivityStarted(activity)
+            onActivityResumed(activity)
+        }
+
+        verifyOrder {
+            testLifecycleObserver.onCreate(any())
+            testLifecycleObserver.onStart(any())
+            testLifecycleObserver.onResume(any())
+        }
+    }
+
+    @Test
+    fun `verify order when the app is foregrounded and backgrounded`() {
+        carAppLifecycleOwner.activityLifecycleCallbacks.apply {
+            val activity: Activity = mockActivity()
+            onActivityCreated(activity, mockk())
+            onActivityStarted(activity)
+            onActivityResumed(activity)
+            onActivityPaused(activity)
+            onActivityStopped(activity)
+            onActivityStarted(activity)
+            onActivityResumed(activity)
         }
 
         verifyOrder {
             testLifecycleObserver.onCreate(any())
             testLifecycleObserver.onStart(any())
             testLifecycleObserver.onResume(any())
+            testLifecycleObserver.onPause(any())
+            testLifecycleObserver.onResume(any())
+        }
+    }
+
+    @Test
+    fun `verify order when the lifecycle is backgrounded and foregrounded`() {
```

Comment:
This test is only being added to show it works with the `verify order when the app is foregrounded and backgrounded` example.

But this test would work without the code change. This code path does not use the `activityLifecycleCallbacks`